webapp/src/app.tsx

```diff
@@ -1856,19 +1862,27 @@ export class ProjectView
         if (this.state.tracing)
             opts.trace = true;
 
-        this.stopSimulator();
+        this.stopSimulator(undefined, true);
+        if (token && token.isCancelled()) return Promise.resolve();
 
         const state = this.editor.snapshotState()
         return compiler.compileAsync(opts)
             .then(resp => {
                 this.clearSerial();
                 this.editor.setDiagnostics(this.editorFile, state)
+                if (token && token.isCancelled()) return;
                 if (resp.outfiles[pxtc.BINARY_JS]) {
-                    simulator.run(pkg.mainPkg, opts.debug, resp, this.state.mute, this.state.highContrast, pxt.options.light, clickTrigger)
-                    this.setState({ running: true, showParts: simulator.driver.runOptions.parts.length > 0 })
+                    if (!token || !token.isCancelled()) {
```

Comment:
this was checked a line ago

Reply:
Yes but it takes a while to run the simulator? Should we check after just in case?